launch/doc/source/conf.py

```diff
@@ -26,9 +26,9 @@
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 #
-# import os
-# import sys
-# sys.path.insert(0, os.path.abspath('.'))
+import os
+import sys
+sys.path.insert(0, os.path.abspath('..'))
```

Comment:
Sorry, I should have asked this on the last round, but I forgot.  Do we also need this change still?  From the comment above, it looks like we might, but on the other hand this only adds in `launch/doc` to the path.  And there don't seem to be any modules in there.  Just wondering what the deal is here.

Reply:
Yes we need this change but maybe I should add a comment explaining the path.

When users' supply custom `conf.py` files, `sphinx_builder.py` will copy+augment this file into the `docs_build` folder. The python interpreter when executing this `conf.py` does not have the modules in its system path. The fix for this this is two step:
1. [Copy](https://github.com/Yadunund/rosdoc2/blob/226f90562be408979902d5102d68f96ced173ffc/rosdoc2/verbs/build/builders/sphinx_builder.py#L612-L615) the pkg's src dir into the `docs_build` folder. This is an extension to the approach [here](https://github.com/ros-infrastructure/rosdoc2/pull/47) which copies files from the `pkg/doc` folder into `docs_build`
2. The change here which adds the copied src files to the system path so that module imports work

Reply:
OK, sounds good.  I'll leave this comment open until we add in that comment, then this will be good to go.

Reply:
Done [68b2466](https://github.com/ros2/launch/pull/697/commits/68b246684528914267df9929d60a4f8b156ba885)

Any tips on getting CI to pass?

Reply:
It looks like there are 2 separate issues.  The first one is with mypy, and the error messages (from https://build.ros2.org/job/Rpr__launch__ubuntu_jammy_amd64/228/consoleFull) look like:

```
09:22:54 198 files checked
09:22:54 2 errors
09:22:54 doc/source/conf.py:104:1: error: Need type annotation for "exclude_patterns" (hint: "exclude_patterns: List[<type>] = ...")
09:22:54 doc/source/conf.py:147:1: error: Need type annotation for "latex_elements" (hint: "latex_elements: Dict[<type>, <type>] = ...")
09:22:54 Found 2 errors in 1 file (checked 198 source files)
```

The second one comes from pep257, and the error messages look like:
```
09:22:54 ./launch/utilities/type_utils.py:380 in public function `normalize_typed_substitution`: D407: Missing dashed underline after section ('Example')
09:22:54 ./launch/utilities/type_utils.py:380 in public function `normalize_typed_substitution`: D412: No blank lines allowed between a section header and its content ('Example')
09:22:54 2 errors
```

Reply:
Thanks! CI passes now. One side effect of leaving the `Example` section header in the docstring of `normalize_type_substitution` is that it generates a sphinx warning (see below). This behavior seems to have been fixed [upstream](https://github.com/sphinx-doc/sphinx/pull/10919) but may not make it to jammy. I think the warning can be ignored since the document still generates successfully with the python code blocks.

```
Examples
--------
/home/yadunund/ws_rolling/docs_build/launch/launch/utilities/type_utils.py:docstring of launch.utilities.type_utils.normalize_typed_substitution:7: WARNING: Unexpected section title.
```